### PR TITLE
Add course landing page

### PR DESCRIPTION
**Pull request by gabrielcaires**
## Introduced changes
* 8 new patterns to compose the new course landing page
* Course landing page pattern
* New template to control a menu variation
* Menu variation that disables the stick and doesn't use a background color
* Support the current sensei message
* Script to move the sensei message to a place after the cover block when the  'Page using transparent menu' is selected.

### Extra Notes: 
1) For some unknown reason, the new page pattern is not generating a preview, as soon we have a solution it will be handled in another PR.

### Testing instructions
Scenario 1: Pattern with the regular template
* Create a new course
* Follow the course creation wizard steps
* Select the `Course Landing Page with cover featured image`  pattern
* Check the layout

Scenario 2: Pattern using the Landing page 
* Create a new course
* Follow the course creation wizard steps
* Select the `Course Landing Page with cover featured image`  pattern
* Select the page template 'Page using transparent menu'
* Publish
* Check if the featured image is above the menu
* Scroll down the page and check that the menu stick is not enabled

Check the mobile version. 

## Videos

** Mobile Version**
[Screen Recording on 2022-10-23 at 11_29_59.webm](https://user-images.githubusercontent.com/38718/197397838-fb9ee4a2-8ef2-458a-b84a-51ff3a4e7d52.webm)

** Desktop Version**
[Screen Recording on 2022-10-23 at 11_32_20.webm](https://user-images.githubusercontent.com/38718/197397929-18bcbff8-35f6-411f-976f-4ed6282df17f.webm)

** Mobile Version**
[Screen Recording on 2022-10-23 at 11_29_59.webm](https://user-images.githubusercontent.com/38718/197397838-fb9ee4a2-8ef2-458a-b84a-51ff3a4e7d52.webm)

## Screenshots

**Desktop Version without featured image:**
Screenshot on 2022-10-23 at 11_18_12

**Desktop Version with featured image:**
Screenshot on 2022-10-23 at 11_22_59

### CODE REVIEW

**Comment by gabrielcaires** [comment]
@Luchadores Do you have any suggestions to handle the mobile version when the course doesn't have a featured image?

**Comment by Luchadores** [comment]
Hey @gabrielcaires 👋 From what I see, this page uses the regular blocks we have in Gutenberg. So I would keep the same behavior they already have for both mobile and desktop.

Let me know if you are referring to something different and if I didn't understand it. Happy to help

**Comment by Luchadores** [comment]
One note: in the case of no featured image, I wouldn't even keep the overlay on top of the Cover. I would remove that completely and keep the cover clean. 
image

image

Note: From what I can see on that specific block, the user cannot really remove the image. They do have the option to replace it. 
image

**Comment by gabrielcaires** [comment]
@Luchadores Are you using WordPress 6.1? It is necessary to see the same options that the user will see when we release our theme.

[Screen Recording on 2022-10-26 at 10_25_08.webm](https://user-images.githubusercontent.com/38718/198038008-5c0f44d4-acd3-4b7c-a844-cd93500ece3f.webm)
As you can see here is possible to remove the image.

I saw that the dark green is caused by an opacity that is applied by default (You can see it on the video), I can set it to zero (no opacity) in order to avoid this division.

**Comment by Luchadores** [comment]
@gabrielcaires - nice, I'm glad we can remove the dark green by setting its opacity to 0. 
 And yes, I'm using WordPress 6.1. Thanks for showing how to remove the image 😃

**Comment by gabrielcaires** [comment]
@Luchadores Oh, I just noticed we are not using the same components.

I am not using it because Media & Text doesn't enable the users to set the Featured Image.
You can download the theme version from this branch here https://github.com/Automattic/sensei-theme/archive/refs/heads/add/course-landing-page.zip if you want.

**Comment by Imran92** [comment]
The contents in the design use 1200px width, we are using 1000px here, so everything looks a bit narrower. For example the top part (It works as expected with the added template)

In Pattern
image

In Design
image

**Comment by Imran92** [comment]
The line height does not match the design here

image

**Comment by Imran92** [comment]
The font for heading is different than the design and the bottom spcing does not look right
image

**Comment by Imran92** [comment]
The line height isn't the same as the design

image

**Comment by Imran92** [comment]
The vertical and horizontal spacings and alignments between these two parts aren't the same as the design

Pattern
image

Design
image

**Comment by Imran92** [comment]
The font size, line height of the text and the space between the author and the text does not match. The Author also has an underline in the design
image

**Comment by Imran92** [comment]
The spacing between these two is 80px in the design, here it is about 120. It needs to be closer

image

**Comment by Imran92** [comment]
The font and the line height does not match with the design
image
 
Design
image

**Comment by Imran92** [comment]
The spacing on top and the line height does not match the design

image

**Comment by Imran92** [comment]
The font here should the Garamond as per design. Also the font size and line height does not match the design

image

**Comment by Imran92** [comment]
This header looks different from the design

Pattern
image


Design
image

**Comment by Imran92** [comment]
This part looks different than the design, I think font weight, spacing etc can be an issue
Pattern
image


Design
image

**Comment by Imran92** [comment]
This portion does not match the design

Pattern
image

Design
image

**Comment by Imran92** [comment]
The image here has a border in the pattern, but the design doesn't

Pattern
image

Design
image

**Comment by Imran92** [comment]
The vertical spacings between these should match the design

Pattern
image

Design
image

**Comment by Imran92** [comment]
The font weight Does not match the design in this part

image

**Comment by Imran92** [comment]
I think this part should be capitalized

Pattern
image

Design

image

**Comment by gabrielcaires** [comment]
> The line height does not match the design here
> 
> image

Fixed here 5b7a1086adb7ab3031d5d50ae8e69b3a38bdcdf8

**Comment by gabrielcaires** [comment]
> The line-height does not match the design here
> 

Updated here  5b7a1086adb7ab3031d5d50ae8e69b3a38bdcdf8

**Comment by gabrielcaires** [comment]
> The font for the heading is different than the design and the bottom spcing does not look right 

Fixed here bb254480a8a7a804cf67dd1d55f78052cb16b375

**Comment by donnapep** [comment]
The pattern name is a bit long. We can probably use "Landing Page" instead.

**Comment by gabrielcaires** [comment]
> The font here should the Garamond as per design. Also the font size and line height does not match the design
> 

Learn block fixed here [e1b9dbd](https://github.com/Automattic/sensei-theme/pull/49/commits/e1b9dbde67430808536f24733df905cb13e5ef57)

**Comment by gabrielcaires** [comment]
@Imran92 I address all of your comments in a big layout revision.
Could you please review it again ?

File: patterns/page-course-with-cover.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+/**
+ * Title: Landing Page
```
Comment (Imran92) [inline patterns/page-course-with-cover.php:4]: Is the title change intentional?

**Comment by Imran92** [comment]
1. The pattern title has changed
image

2. Is this image expected to be loaded automatically with the pattern? Not doing that for me
image

3. The pattern should match the design here
Design:
image
Pattern:
image

4. The font size is 28px here, in design it is 24px
image

5. Letter spacing, line height differs from the design here
image

6. The font weight is 500 in the pattern, in design it is 590
image

7. This part does not look like the design in desktop with same text as the design

Design
image

Pattern
image

File: patterns/page-course-with-cover.php (hunk 1 above)
Comment (gabrielcaires) [reply patterns/page-course-with-cover.php:4]: It was requested [here](https://github.com/Automattic/sensei-theme/pull/49#issuecomment-1295159999)

**Comment by donnapep** [comment]
We need to move this PR to Sensei, and make these patterns available when Sensei is installed, not when the theme is installed.

**Comment by gabrielcaires** [comment]
Moving it to draft to avoid confusions

**Comment by donnapep** [comment]
Closing as repo is being archived.

**Pull request closed without merging**